src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java

```diff
@@ -0,0 +1,666 @@
+    {
+        super.serializeToView(buf);
+        buf.writeBoolean(assignManually);
+        buf.writeBoolean(retrieveOnLowHealth);
+        buf.writeBoolean(patrolManually);
+        buf.writeInt(task.ordinal());
+        buf.writeInt(job == null ? -1 : job.ordinal());
+        buf.writeInt(patrolTargets.size());
+
+        for (final BlockPos pos : patrolTargets)
+        {
+            BlockPosUtil.writeToByteBuf(buf, pos);
+        }
+
+        BlockPosUtil.writeToByteBuf(buf, guardPos);
+    }
+
+    /**
+     * Getter for the bonus vision.
+     *
+     * @return an integer for the additional range.
+     */
+    public int getBonusVision()
+    {
+        if (getBuildingLevel() <= MAX_VISION_BONUS_MULTIPLIER)
+        {
+            return getBuildingLevel() * VISION_BONUS;
+        }
+        return MAX_VISION_BONUS_MULTIPLIER * VISION_BONUS;
+    }
+
+    /**
+     * Check if the guard should retrieve on low health.
+     *
+     * @return true if so.
+     */
+    public boolean shallRetrieveOnLowHealth()
+    {
+        return retrieveOnLowHealth;
+    }
+
+    /**
+     * Set if the guard should retrieve on low health.
+     *
+     * @param retrieveOnLowHealth state to set.
+     */
+    public void setRetrieveOnLowHealth(final boolean retrieveOnLowHealth)
+    {
+        this.retrieveOnLowHealth = retrieveOnLowHealth;
+        this.markDirty();
+    }
+
+    /**
+     * Check if the guard should patrol manually (Use defined values).
+     *
+     * @return true if so.
+     */
+    public boolean shallPatrolManually()
+    {
+        return patrolManually;
+    }
+
+    /**
+     * Set if the guard should patrol manually.
+     *
+     * @param patrolManually the state to set.
+     */
+    public void setPatrolManually(final boolean patrolManually)
+    {
+        this.patrolManually = patrolManually;
+        this.markDirty();
+    }
+
+    /**
+     * Getter of the position the gurd should guard.
+     *
+     * @return the blockPos position.
+     */
+    public BlockPos getGuardPos()
+    {
+        return guardPos;
+    }
+
+    /**
+     * Set if the jobs should be assigned manually.
+     *
+     * @param assignManually the state to set.
+     */
+    public void setAssignManually(final boolean assignManually)
+    {
+        this.assignManually = assignManually;
+        this.markDirty();
+    }
+
+    /**
+     * Possible job/AI.
+     */
+    public enum GuardJob
+    {
+        KNIGHT,
+        RANGER,
+    }
+
+    /**
+     * Possible tasks.
+     */
+    public enum Task
+    {
+        FOLLOW,
+        GUARD,
+        PATROL
+    }
+
+    /**
+     * The name of the guards's job.
+     *
+     * @return The name of the guards's job.
+     */
+    @NotNull
+    @Override
+    public String getJobName()
+    {
+        return GUARD_TOWER;
+    }
+
+    /**
+     * The client view for the baker building.
+     */
+    public static class View extends AbstractBuildingWorker.View
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 58 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)